docs/content/concepts/partitions-schedules-sensors/asset-sensors.mdx

```diff
@@ -218,9 +248,32 @@ def repository_1():
 
 This sensor will materialize `asset_b` whenever any of its parents materialize, and it will materialize `asset_c` whenever any of its parents materialize.
 
+### Freshness-based scheduling
+
+<Note>
+  This feature is currently <strong>Experimental</strong>.
+</Note>
+
+Setting a `FreshnessPolicy` on an asset specifies how up-to-date the data in that asset is expected to be. The asset reconciliation sensor created using `build_asset_reconciliation_sensor` will kick off runs to keep that data in line with the defined policy.
+
+To minimize unnecessary computation, the sensor will take into account a few bits of information when determining if it needs to kick off a run for a given asset:
+
+- A timestamp of the upstream data that is incorporated into the current state of the asset. This is calculated by finding the most recent materialization of that asset, then recursively finding the most recent materializations of its parents that happened before that event. The minimum timestamp found this way will be used as the current data time.
+- Any in-progress runs that will update that asset in the near future. If an in-progress run will update an asset to an acceptable data time, then a new run will not be kicked off.
+- The constraints specifying what upstream data must be incorporated into the state of an asset by certain times. If waiting some amount of time to kick off a materialization of an asset will allow multiple constraints to be satisfied at once, then the asset will not be materialized on that tick.
+
+The sensor will only kick off runs for assets specified in the `asset_selection` argument of `build_asset_reconciliation_sensor`. If an asset in this selection is downstream of an asset outside of the selection, the sensor will not kick off a run of the selected asset until a materialization event for the upstream asset is created.
```

Comment:
I don't follow this - how would the materialization event of the upstream asset be created? Would the freshness policy be violated if the upstream is never updated?